utilities/src/main/kotlin/org/jetbrains/kotlin/cli/utilities/InteropCompiler.kt

```diff
@@ -28,24 +28,21 @@ fun invokeInterop(flavor: String, args: Array<String>): Array<String>? {
 
     val buildDir = File("$outputFileName-build")
     val generatedDir = File(buildDir, "kotlin")
-    val nativesDir = File(buildDir, "natives")
+    val nativesDir = File(buildDir,"natives")
     val manifest = File(buildDir, "manifest.properties")
-    val additionalArgs = listOf(
-            "-generated", generatedDir.path,
-            "-natives", nativesDir.path,
-            "-flavor", flavor
-    )
     val cstubsName ="cstubs"
     val libraries = arguments.library
     val repos = arguments.repo
     val targetRequest = if (arguments is CInteropArguments) arguments.target
         else (arguments as JSInteropArguments).target
     val target = PlatformManager().targetManager(targetRequest).target
 
-    val cinteropArgsToCompiler = interop(flavor, args + additionalArgs,
+    val cinteropArgsToCompiler = interop(flavor, args,
             listOfNotNull(
```

Comment:
All these maps and unsafe casts are looking very python-ish :)
There are better solution in a typed world. For example, you can create a `data` class.